lib/menu.c

```diff
@@ -771,14 +771,15 @@ bm_menu_filter(struct bm_menu *menu)
         list_free_list(&menu->filtered);
         free(menu->old_filter);
         menu->old_filter = NULL;
+
+        menu->filtered.count = menu->items.count;
```

Comment:
Please don't do this as it will cause inconsistent state. Instead in the `bm_cairo_paint` function assign to `filtered_item_count` like this: `(menu->filter ? menu->filtered.count : total_item_count)`

Reply:
Ok, working on it. Thought it wouldn't really work out in the end but it fixed the "bug".